epochs/hippieeg/workflow/Snakefile

```diff
@@ -0,0 +1,109 @@
+#---- begin snakebids boilerplate ----------------------------------------------
+
+import snakebids
+from snakebids import bids
+from os.path import join
+from itertools import compress
+import numpy as np
+import sys
+sys.path.append('/scratch/mcesped/HippiEEGAtlas/epochs/hippieeg/code')
+from file_manager import bids_folder
+
+
+configfile: workflow.source_path('../config/snakebids.yml')
+
+# Get input wildcards
+inputs = snakebids.generate_inputs(
+    bids_dir=config["bids_dir"],
+    pybids_inputs=config["pybids_inputs"],
+    pybids_database_dir=config.get("pybids_db_dir"),
+    pybids_reset_database=config.get("pybids_db_reset"),
+    derivatives=config.get("derivatives", None),
+    participant_label=config.get("participant_label", None),
+    exclude_participant_label=config.get("exclude_participant_label", None),
+    use_bids_inputs=True,
+)
+subj_session_list = list(zip(inputs.zip_lists['bold']['subject'],inputs.zip_lists['bold']['session']))
+# Annotation files
+inputs_annot = snakebids.generate_inputs(
+    bids_dir=config["bids_dir"],
+    pybids_inputs=config["pybids_inputs_annot"],
+    pybids_database_dir=config.get("pybids_db_dir"),
+    pybids_reset_database=config.get("pybids_db_reset"),
+    derivatives=config.get("derivatives", None),
+    participant_label=config.get("participant_label", None),
+    exclude_participant_label=config.get("exclude_participant_label", None),
+    use_bids_inputs=True,
+)
+# TSV files input
+input_tsv = snakebids.generate_inputs(
+    bids_dir=config["bids_dir_tsv"],
+    pybids_inputs=config["pybids_inputs_tsv"],
+    pybids_database_dir=config.get("pybids_db_dir"),
+    pybids_reset_database=config.get("pybids_db_reset"),
+    derivatives=config.get("derivatives", None),
+    participant_label=config.get("participant_label", None),
+    exclude_participant_label=config.get("exclude_participant_label", None),
+    use_bids_inputs=True,
+)
+
+#this adds constraints to the bids naming
+wildcard_constraints:  **snakebids.get_wildcard_constraints(config['pybids_inputs'])
+# print(inputs.path)
+# print(inputs.zip_lists['bold']['run'])
+# print(inputs.zip_lists['bold']['task'])
+# print(subj_session_list)
+# print(inputs.zip_lists['bold']['subject'])
+# print(inputs.zip_lists['bold']['session'])
+
+#---- end snakebids boilerplate ------------------------------------------------
+rule epochs:
+    input: 
+        edf = lambda wildcards: expand(inputs.path['bold'], zip,
```

Comment:
I doubt that you need to use this kind of complicated code to fill wildcards in the input. it should be automatically populated from the target rule. The only variable you should want to expand over is the the one you want to aggregate which if I'm not mistaken is subjects.